### PR TITLE
added chardet encoding detection in filer_check

### PR DESCRIPTION
**Pull request by jarriagad**
Related to [Zayo email fails parser on email signature](https://github.com/networktocode/circuit-maintenance-parser/issues/175)

changes made to circuit_maintenance_parser/provider.py

Adds chardet.detect method before decoding data_part.content

### CODE REVIEW

**Comment by chadell** [comment]
thanks for your PR @jarriagad !

could you run a `poetry lock` to add the `chardet` library to the solved dependencies?

**Review by scetron — APPROVED** [review]
+1 to Christian's Comment. Thank you for your contribution.

**Comment by jarriagad** [comment]
@chadell Thanks for the quick response on the PR!
I ran the poetry commands and think I uploaded the test data correctly.

Let me know if anything else needs changes.

Thanks!

**Comment by chadell** [comment]
@jarriagad , it seems that a dev dependency for `mypy` will be needed: `types-chardet`

**Comment by jarriagad** [comment]
@chadell I've added the types-chardet dependency. The IPs have been manually anonymized in the test data

**Comment by chadell** [comment]
Small issue with `pylint`. You can run the tests locally, with `invoke test`.

**Comment by jarriagad** [comment]
@chadell I fixed the error noted in the pipeline and have pushed those changes but when I run `invoke tests --local` i get the error

```
would reformat venv/lib/python3.10/site-packages/yamllint/rules/trailing_spaces.py
would reformat venv/lib/python3.10/site-packages/yamllint/rules/truthy.py

Oh no! 💥 💔 💥
2486 files would be reformatted, 1520 files would be left unchanged, 5 files would fail to reformat.
```

Not sure what I am doing wrong.

**Comment by chadell** [comment]
> @chadell I fixed the error noted in the pipeline and have pushed those changes but when I run `invoke tests --local` i get the error
> 
> ```
> would reformat venv/lib/python3.10/site-packages/yamllint/rules/trailing_spaces.py
> would reformat venv/lib/python3.10/site-packages/yamllint/rules/truthy.py
> 
> Oh no! 💥 💔 💥
> 2486 files would be reformatted, 1520 files would be left unchanged, 5 files would fail to reformat.
> ```
> 
> Not sure what I am doing wrong.

this happens because you created your virtualenv in the same path `venv/`,  and it get analyised by the linter

File: pyproject.toml [hunk 1]
```diff
@@ -32,6 +32,8 @@ lxml = "^4.6.2"
 geopy = "^2.1.0"
 tzwhere = "^3.0.3"
 backoff = "^1.11.1"
+chardet = "^5"
+types-chardet = "^5.0.4"
```
Comment (chadell) [inline pyproject.toml:36]: We should move this to `tool.poetry.dev-dependencies`, because it's only needed for typing analysis

Comment (jarriagad) [reply pyproject.toml]: This have been updated in the latest push

Also tested locally and got the following results
```
LOCAL - Running command black --check --diff .
All done! ✨ 🍰 ✨
41 files would be left unchanged.
LOCAL - Running command flake8 .
LOCAL - Running command find . -name "*.py" | xargs pylint
************* Module tasks
tasks.py:4:0: W0402: Uses of a deprecated module 'distutils.util' (deprecated-module)

------------------------------------------------------------------
Your code has been rated at 9.99/10 (previous run: 9.99/10, +0.00)
```

**Review by chadell — APPROVED** [review]
LGTM

**Comment by chadell** [comment]
thanks @jarriagad for your first contribution!

**Pull request merged**